ext/random/php_random.h

```diff
@@ -71,9 +71,9 @@ typedef struct _php_random_status_state_combinedlcg {
 } php_random_status_state_combinedlcg;
 
 typedef struct _php_random_status_state_mt19937 {
-	uint32_t state[MT_N];
 	uint32_t count;
-	uint8_t mode;
+	enum php_random_mt19937_mode mode;
+	uint32_t state[624];
```

Comment:
Does this order increase the size of the struct or not?

Reply:
It does not. This is the previous ordering:

```
struct _php_random_status_state_mt19937 {
        uint32_t                   state[624];           /*     0  2496 */
        /* --- cacheline 39 boundary (2496 bytes) --- */
        uint32_t                   count;                /*  2496     4 */
        uint8_t                    mode;                 /*  2500     1 */

        /* size: 2504, cachelines: 40, members: 3 */
        /* padding: 3 */
        /* last cacheline: 8 bytes */
};
```

Note how the struct has a padding of 3 bytes to make its size a multiple of 4, effectively increasing `mode`'s size to 4 bytes. Moving `mode` into the middle of the struct would result in a 3 byte hole, but changing it to an `enum` increased its size to 4 bytes anyway.

An any case, total size is unchanged:

```
struct _php_random_status_state_mt19937 {
        uint32_t                   count;                /*     0     4 */
        enum php_random_mt19937_mode mode;               /*     4     4 */
        uint32_t                   state[624];           /*     8  2496 */

        /* size: 2504, cachelines: 40, members: 3 */
        /* last cacheline: 8 bytes */
};
```